### ISSUE TITLE
layers for `.raw`

### ISSUE THREAD

**Issue by grst** [scverse/anndata#96]
I tried to store multiple layers in the `.raw` adata (I want to keep both raw UMI counts and normalized, log-transformed counts for plotting). That doesn't seem to be supported at the moment. 

1) This seems pretty straightforward to implement. Add a `_layers` attribute, add a `@property` and update the `__getitem__` function here: https://github.com/theislab/anndata/blob/2ab6e8edee811990048c66e9afdc507faecb7bb6/anndata/base.py#L517

I'm just not sure how I treat the case when the anndata is backed. 

I could try to put a PR together. 

2) Why not just using a normal `adata` object for raw and simply remove the `obs`, `obsm` and `uns` attributes when setting it? Like that you wouldn't have to maintain a separate class with independent logic and also things like layers or `adata.raw.shape` would simply work.

**Comment by LuckyMD** [comment]
Hi,
I just noticed this conversation, and I hope you don't mind if I butt in. I like the idea of potentially doing completely without `.raw`. I imagine it may be quite difficult to change the code in this way (and backward compatability would not really be possible... scanpy 2.0?), but the `.raw` setup with potentially different matrix dimensions than the `AnnData.X` "layer" has caused me quite some frustration. I think it has also lead to quite a bit of code complexity that could be avoided.

**Comment by falexwolf** [comment]
Thank you for your considerations! Let's keep it on the list of future considerations!

**Comment by fidelram** [comment]
@LuckyMD @falexwolf About future considerations I think that the `.raw` model can be used for multi-modal data (e.g. CITE-seq) which, I expect, will become more common. 

The idea behind `.raw` is that `obs` is shared with `adata.X` but not `var`, which is the basis for multi-modal data. 

Thus, instead of removing `.raw`, I think that it can be repurposed.

**Comment by LuckyMD** [comment]
That's a really good point actually @fidelram. I wasn't thinking about multi-modal data with different `.var` at all. For multiple joint profiling data we would actually need an extension of `.raw` rather than the opposite. I feel like that would change the idea behind what `.raw` is though... it would need to be more general and not just a default for statistical testing and plotting of expression values. Essentially we'd need `.layers` with non-shared `.var`.

**Comment by fidelram** [comment]
Indeed. As I see it, the basic structure is: layers share obs and var,
multi_modals share obs only. I am hoping that most of the code for slicing
and saving from .raw can be re-used for multi-modal.
